src/main/java/pl/codilingus/library/validation/OrderValidator.java

```diff
@@ -0,0 +1,45 @@
+package pl.codilingus.library.validation;
+import org.springframework.beans.factory.annotation.Autowired;
+import pl.codilingus.library.Book;
+import pl.codilingus.library.Order;
+import pl.codilingus.library.controllers.OrderController;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class OrderValidator {
+
+    @Autowired
+    private OrderController orderController;
+
+    private boolean validateOrderingMonth(Order newOrder){
+        List<Order> listOfOrders = orderController.getAllOrders();
```

Comment:
How about moving `List<Order> listOfOrders = orderController.getAllOrders();` above to `validationMechanism` adn pass it as a parameter to these private methods? Then we could query database for list of orders only once and then reuse this list.